yield_plugin/src/yield_plugin.cpp

```diff
@@ -600,92 +597,154 @@ namespace yield_plugin
-      if (config_.enable_adjustable_gap)
-      {
-        // check if a digital_gap is available
-        double digital_gap = check_traj_for_digital_min_gap(original_tp);
-        RCLCPP_DEBUG_STREAM(nh_->get_logger(),"digital_gap: " << digital_gap);
-        // if a digital gap is available, it is replaced as safety gap
-        safety_gap = std::max(safety_gap, digital_gap);
-      }
-      // safety gap is implemented
-      double goal_pos = x_lead - safety_gap;
+    lanelet::BasicPoint2d point(original_tp.trajectory_points[0].x,original_tp.trajectory_points[0].y);
+    double vehicle_downtrack = wm_->routeTrackPos(point).downtrack;
 
-      if (goal_velocity <= config_.min_obstacle_speed){
-        RCLCPP_WARN(nh_->get_logger(),"The obstacle is not moving, goal velocity is set to 0");
-        goal_velocity = 0.0;
-      }
+    RCLCPP_DEBUG_STREAM(nh_->get_logger(),"vehicle_downtrack: " << vehicle_downtrack);
 
-      double initial_time = 0;
-      double initial_pos = 0.0; //relative initial position (first trajectory point)
+    RCLCPP_WARN_STREAM(nh_->get_logger(),"Collision Detected!");
 
-      double initial_accel = 0;
-      double goal_accel = 0;
+    lanelet::BasicPoint2d point_o(earliest_collision_obj.pose.pose.position.x, earliest_collision_obj.pose.pose.position.y);
+    double object_downtrack = wm_->routeTrackPos(point_o).downtrack;
 
-      double delta_v_max = fabs(earliest_collision_obj.value().velocity.twist.linear.x - max_trajectory_speed(original_tp.trajectory_points));
-      // reference time, is the maximum time available to perform object avoidance (length of a trajectory)
-      double t_ref = (rclcpp::Time(original_tp.trajectory_points[original_tp.trajectory_points.size() - 1].target_time).seconds() - plan_start_time.seconds());
-      // time required for comfortable deceleration
-      double t_ph = config_.acceleration_adjustment_factor * delta_v_max / config_.yield_max_deceleration;
+    RCLCPP_DEBUG_STREAM(nh_->get_logger(),"object_downtrack: " << object_downtrack);
 
-      // planning time for object avoidance
-      double tp = 0;
+    double x_lead = std::max(0.0, object_downtrack - vehicle_downtrack);
+    RCLCPP_DEBUG_STREAM(nh_->get_logger(),"x_lead: " << x_lead);
 
-      if(t_ph > config_.tpmin && t_ref > t_ph)
-      {
-        tp = t_ph;
-      }
-      else if(t_ph < config_.tpmin)
-      {
-        tp = config_.tpmin;
-      }
-      else
-      {
-        tp = t_ref;
-      }
+    double goal_velocity = get_object_velocity_along_trajectory(earliest_collision_obj.velocity.twist, original_tp, earliest_collision_time);
+    RCLCPP_DEBUG_STREAM(nh_->get_logger(),"object's speed along trajectory at collision: " << goal_velocity);
 
-      RCLCPP_DEBUG_STREAM(nh_->get_logger(),"Object avoidance planning time: " << tp);
+    // roadway object position
+    double gap_time = std::max(0.0, x_lead - config_.x_gap)/initial_velocity;
```

Comment:
Is `gap_time` the duration it takes to travel the gap distance? If so, this calculation looks wrong. It's subtracting the current gap (object downtrack - vehicle downtrack) from the minimum allowed gap (from the configuration). It seems to be calculating the time to travel the distance between current gap and minimum allowed gap.

Reply:
Yeah I think the 2nd is the intention - calculating time it takes to travel from having current gap to having minimum allowed gap from the last fixed point of the front vehicle. I renamed to gap_time_until_min_gap_distance. lmk if it works